features/step_definitions/two_factor_steps.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+When /^I scan the QR code and fill in a valid TOTP token for "([^"]*)"$/ do |email|
+  @me = find_user email
+  fill_in "user_code", with: @me.current_otp
+end
+
+When /^I fill in a valid TOTP token for "([^"]*)"$/ do |username|
+  @me = find_user username
+  fill_in "user_otp_attempt", with: @me.current_otp
+end
+
+When /^I fill in an invalid TOTP token$/ do
+  fill_in "user_otp_attempt", with: "c0ffee"
+end
+
+When /^I fill in a valid TOTP token to deactivate for "([^"]*)"$/ do |username|
+  @me = find_user username
+  fill_in "two_factor_authentication_code", with: @me.current_otp
+end
+
+When /^I fill in an invalid TOTP token to deactivate$/ do
+  fill_in "two_factor_authentication_code", with: "c0ffee"
+end
+
+When /^I fill in a recovery code from "([^"]*)"$/ do |username|
+  @me = find_user username
+  @codes = @me.generate_otp_backup_codes!
+  @me.save!
+  fill_in "user_otp_attempt", with: @codes.first
+end
+
+When /^I fill in a recovery code to deactivate from "([^"]*)"$/ do |username|
+  @me = find_user username
+  @codes = @me.generate_otp_backup_codes!
+  @me.save!
+  fill_in "two_factor_authentication_code", with: @codes.first
+end
+
+When /^I click on OK$/ do
+  find(".btn-primary", match: :first).click
```

Comment:
Steps are available everywhere, I feel like this is too generic and can cause problems. We already have generic steps like "I validate the modal". Can they be used there? If no, can we make that step more specific?